pages/_app.tsx

```diff
@@ -26,11 +28,27 @@ function App({ Component, pageProps }: AppProps) {
     setMounted(true);
   }, []);
 
-  const router = useRouter();
+  // An ugly, terrible and sad hack to scroll the page to the
+  // anchor location when present. The reason this stopped working is
+  // due to the dark theme hack. :facepalm:
+  React.useEffect(() => {
+    if (mounted) {
+      const [_, hashLocation] = router.asPath.split('#');
+      if (hashLocation) {
+        const anchor = document.querySelector(`#${hashLocation}`);
+        const scrollMargin = 20;
+        const distanceToScroll =
+          window.pageYOffset + anchor.getBoundingClientRect().top - scrollMargin;
```

Comment:
```suggestion
          window.pageYOffset + anchor.getBoundingClientRect().top - scrollMargin;
```
Don't think this needs `getBoundingClientRect`, just `offsetTop`

Reply:
I think `offsetTop` only gives you the distance to its nearest parent, I want it from the entire page. 

I think to do it properly wth `offsetTop` I'd need to traverse up the tree in a `while` loop, no?